packages/player/src/components/shortcut.tsx

```diff
@@ -0,0 +1,370 @@
+import * as React from 'react'
+import {VideoContext, VideoStateContext} from './player'
+import {useSelector} from '@xstate/react'
+import {State} from 'xstate'
+
+type ShortcutProps = {
+  clickable?: boolean
+  dblclickable?: boolean
+  shortcuts?: any[]
+}
+
+const selectHasStarted = (state: {context: VideoStateContext}) =>
+  state.context.hasStarted || false
+
+const selectIsActive = (state: {context: VideoStateContext}) =>
+  state.context.isActive || false
+
+const selectReadyState = (state: {context: VideoStateContext}) =>
+  state.context.readyState ?? -1
+
+const selectVolume = (state: {context: VideoStateContext}) =>
+  state.context.volume ?? 0.8
+
+const selectCurrentTime = (state: {context: VideoStateContext}) =>
+  state.context.video?.currentTime ?? 0
+
+const selectDuration = (state: {context: VideoStateContext}) =>
+  state.context.video?.duration ?? 0
+
+const selectIsPaused = (state: State<VideoStateContext>) =>
+  state.matches('ready.paused')
+
+const selectPlaybackRate = (state: State<VideoStateContext>) =>
+  state.context.playbackRate ?? 1.0
+
+export const Shortcut: React.FC<ShortcutProps> = ({
+  clickable = false,
+  dblclickable = false,
+  ...props
+}) => {
+  const {videoService} = React.useContext(VideoContext)
+  const hasStarted = useSelector(videoService, selectHasStarted)
+  const isActive = useSelector(videoService, selectIsActive)
+  const readyState = useSelector(videoService, selectReadyState)
+  const volume = useSelector(videoService, selectVolume)
+  const duration = useSelector(videoService, selectDuration)
+  const currentTime = useSelector(videoService, selectCurrentTime)
+  const paused = useSelector(videoService, selectIsPaused)
+  const playbackRate = useSelector(videoService, selectPlaybackRate)
```

Comment:
it ends up being pretty verbose, but also very clear

selectors could be reused